api/src/main/java/org/openmrs/api/db/hibernate/HibernateOrderDAO.java

```diff
@@ -692,53 +777,81 @@ public void purgeOrderType(OrderType orderType) {
 	 */
 	@Override
 	public List<OrderType> getOrderSubtypes(OrderType orderType, boolean includeRetired) {
-		Criteria criteria = sessionFactory.getCurrentSession().createCriteria(OrderType.class);
-		criteria.add(Restrictions.eq("parent", orderType));
+		Session session = sessionFactory.getCurrentSession();
+		CriteriaBuilder cb = session.getCriteriaBuilder();
+		CriteriaQuery<OrderType> cq = cb.createQuery(OrderType.class);
+		Root<OrderType> root = cq.from(OrderType.class);
+
 		if (!includeRetired) {
-			criteria.add(Restrictions.eq("retired", false));
+			cq.where(cb.equal(root.get("parent"), orderType), cb.isFalse(root.get("retired")));
```

Comment:
Shouldn't `cq.where(cb.equal(root.get("parent")` be outside the if else clause?

Reply:
Moved to outside